src/Icicle/Source/ToCore/ToCore.hs

```diff
@@ -637,19 +638,21 @@ convertReduce xx
 
 
  | Case (Annot { annAnnot = ann, annResult = retty }) scrut patalts <- xx
- = do   scrut' <- convertReduce scrut
-        let (pats,alts) = unzip patalts
-        alts'  <- mapM convertReduce alts
+ = do   let (pats,alts) = unzip patalts
+
+        scrut' <- convertReduce scrut
+        pats'  <- convertCaseFreshenPats pats
+        alts'  <- mapM convertExp alts
+        alts'' <- hoist runFreshIdentity $ lift $ mapM (ANormal.anormal ()) alts'
```

Comment:
yeah, you're right. I don't know what I was thinking... a-normal shouldn't change whether the generated Core typechecks.